src/Common/src/System/Net/Logging/NetEventSource.cs

```diff
@@ -0,0 +1,266 @@
+                dataDesc[1].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[1].Size = (method.Length + 1) * sizeof(char);
+                dataDesc[2].DataPointer = (IntPtr)(arg3Ptr);
+                dataDesc[2].Size = (message.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&objHash);
+                dataDesc[3].Size = sizeof(int);
+                dataDesc[4].DataPointer = (IntPtr)(arg4Ptr);
+                dataDesc[4].Size = (stackTrace.Length + 1) * sizeof(char);
+                dataDesc[5].DataPointer = (IntPtr)(&componentType);
+                dataDesc[5].Size = sizeof(int);
+                WriteEventCore(CRITICALEXCEPTION_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void PrintError(ComponentType componentType, string msg)
+        {
+            if (msg == null)
+            {
+                return;
+            }
+            s_log.CriticalError(LoggingHash.GetObjectName(msg), "", "", 0, componentType);
+        }
+
+        [NonEvent]
+        internal static void PrintError(ComponentType componentType, object obj, string method, string msg)
+        {
+            s_log.CriticalError(LoggingHash.GetObjectName(msg),
+                                LoggingHash.GetObjectName(method),
+                                LoggingHash.GetObjectName(obj),
+                                LoggingHash.HashInt(obj),
+                                componentType);
+        }
+
+        [Event(CRITICALERROR_ID, Keywords = Keywords.Default,
+Level = EventLevel.Critical)]
+        internal unsafe void CriticalError(string message, string method, string objName, int objHash, ComponentType componentType)
+        {
+            const int SIZEDATA = 5;
+            fixed (char* arg1Ptr = message, arg2Ptr = method, arg3Ptr = objName)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)arg1Ptr;
+                dataDesc[0].Size = (message.Length + 1) * sizeof(char); // Size in bytes, including a null terminator. 
+                dataDesc[1].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[1].Size = (method.Length + 1) * sizeof(char);
+                dataDesc[2].DataPointer = (IntPtr)(arg3Ptr);
+                dataDesc[2].Size = (objName.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&objHash);
+                dataDesc[3].Size = sizeof(int);
+                dataDesc[4].DataPointer = (IntPtr)(&componentType);
+                dataDesc[4].Size = sizeof(int);
+                WriteEventCore(CRITICALERROR_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        public class Keywords
+        {
+            public const EventKeywords Default = (EventKeywords)0x0001;
+            public const EventKeywords Debug = (EventKeywords)0x0002;
```

Comment:
Can we extract this class and reuse it instead of defining it for each eventsource?
I noticed that `Debug` isn't used in this event-source but it's still defined (to keep numbering).